File: src/libs/ThirdParty.CSSuppliers/ExpediaRapid/SerializableClasses/Prebook/PrebookResponse.cs
namespace ThirdParty.CSSuppliers.ExpediaRapid.SerializableClasses.Prebook
{
    using System;
    using System.Collections.Generic;
    using Newtonsoft.Json;

    public class PrebookResponse : IExpediaRapidResponse
    {

        [JsonProperty("occupancy_pricing")]
        public Dictionary<string, OccupancyRoomRate> OccupancyRoomRates { get; set; }

        [JsonProperty("links")]
        public Dictionary<string, Link> Links { get; set; } = new Dictionary<string, Link>();

        public bool IsValid(string responseString, int statusCode)
        {

            if (!string.IsNullOrWhiteSpace(responseString))
            {
                try
                {

                    JsonConvert.DeserializeObject<PrebookResponse>(responseString);
                    return true;
                }

                catch (Exception ex)
                {
                    return false;
                }
            }
            return false;
        }
    }

}

File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Prebook/PrebookResponse.vb
Imports Newtonsoft.Json.Linq
Imports Intuitive.Net
Imports Newtonsoft.Json

Namespace ExpediaRapid.SerializableClasses.Prebook

    Public Class PrebookResponse
        Implements IExpediaRapidResponse

        <JsonProperty("occupancy_pricing")>
        Public Property OccupancyRoomRates As Dictionary(Of String, OccupancyRoomRate)

        <JsonProperty("links")>
        Public Property Links As New Dictionary(Of String, Link)

        Public Function IsValid(responseString As String, statusCode As Integer) As Boolean Implements IExpediaRapidResponse.IsValid

            If Not String.IsNullOrWhiteSpace(responseString) Then
                Try

                    JsonConvert.DeserializeObject(Of PrebookResponse)(responseString)
                    Return True

                Catch ex As Exception
                    Return False
                End Try
            End If
            Return False
        End Function
    End Class

End Namespace